### PR TITLE
Unit test extract journal info

### PR DESCRIPTION
**Pull request by rikirenz**
This PR should add the test for the extract_journal_info function and also fix the bug that the test found.

### CODE REVIEW

File: tests/unit/refextract/test_tasks.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    eng = DummyEng()
+
+    extract_journal_info(obj, eng)
+
+    pub_info = obj.data['publication_info'][0]
+
+    assert '082102' == pub_info.get('artid')
+    assert '55' == pub_info.get('journal_volume')
+    assert 'J. Math. Phys.' == pub_info.get('journal_title')
+    assert 2014 == pub_info.get('year')
```
Comment (jacquerie) [inline tests/unit/refextract/test_tasks.py]: You can just write
```python
assert obj.data['publication_info'][0] == {
    'artid': '082102',
    'journal_title': 'J. Math. Phys.',
    'journal_volume': '55',
    'year': 2014,
}
```

File: tests/unit/refextract/test_tasks.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    pass
+
+
+def test_extract_journal_info():
+
+    data = {
+        'publication_info': [{'pubinfo_freetext': 'J. Math. Phys. 55, 082102 (2014)'}]
+    }
+
+    obj = StubObj(data)
```
Comment (jacquerie) [inline tests/unit/refextract/test_tasks.py]: IMO it's ok to have
```python
obj = StubObj({
    'publication_info': [
        {'pubinfo_freetext': 'J. Math. Phys. 55, 082102 (2014)'},
    ],
})
```

File: tests/unit/refextract/test_tasks.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    eng = DummyEng()
+
+    extract_journal_info(obj, eng)
```
Comment (jacquerie) [inline tests/unit/refextract/test_tasks.py]: Here you should also test the return value (`None`) of the task, because I believe that if it's different from `None` it's used by the next task as the input (I might be wrong on this):
```python
assert extract_journal_info(obj, eng) is None
```

File: tests/unit/refextract/test_tasks.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014, 2015, 2016 CERN.
```
Comment (jacquerie) [inline tests/unit/refextract/test_tasks.py]: Wrong year.

File: tests/unit/refextract/test_tasks.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+from __future__ import absolute_import, division, print_function
+
+from inspirehep.modules.refextract.tasks import extract_journal_info
+
+
+class StubObj(object):
+    def __init__(self, data):
+        self.data = data
+        self.extra_data = {}
+        self.files = {}
```
Comment (jacquerie) [inline tests/unit/refextract/test_tasks.py]: Kind of weird, but ok.

File: inspirehep/modules/refextract/tasks.py [hunk 6]
```diff
@@ -64,9 +64,9 @@ def extract_journal_info(obj, eng):
                     "title"
                 )
             if "year" in extracted_publication_info:
-                pubnote["year"] = extracted_publication_info.get(
+                pubnote["year"] = int(extracted_publication_info.get(
                     "year"
-                )
+                ))
```
Comment (jacquerie) [inline inspirehep/modules/refextract/tasks.py]: I see that you steered clear from doing the refactoring this task needs : )
